packages/analytics/src/helpers.ts

```diff
@@ -80,7 +77,7 @@ export function insertScriptTag(dataLayerName: string): void {
 export function getOrCreateDataLayer(dataLayerName: string): DataLayer {
   // Check for existing dataLayer and create if needed.
   let dataLayer: DataLayer = [];
-  if (hasDataLayer(dataLayerName)) {
+  if (Array.isArray(window[dataLayerName])) {
     dataLayer = window[dataLayerName] as DataLayer;
   } else {
     dataLayer = window[dataLayerName] = [];
```

Comment:
Since dataLayer is already initialized with `[]`, you can avoid one memory allocation by `window[dataLayerName] = dataLayer`

Reply:
Done.